kazoo/protocol/connection.py

```diff
@@ -448,13 +468,35 @@ def _read_response(self, header, buffer, offset):
 
                 async_object.set(response)
 
-            # Determine if watchers should be registered
-            watcher = getattr(request, "watcher", None)
-            if not client._stopped.is_set() and watcher:
-                if isinstance(request, (GetChildren, GetChildren2)):
-                    client._child_watchers[request.path].add(watcher)
-                else:
-                    client._data_watchers[request.path].add(watcher)
+            # Determine if watchers should be registered or unregistered
+            if not client._stopped.is_set():
+                watcher = getattr(request, "watcher", None)
+                if watcher:
+                    if isinstance(request, AddWatch):
+                        if request.mode == AddWatchMode.PERSISTENT:
+                            client._persistent_watchers[request.path].add(
+                                watcher
+                            )
+                        elif request.mode == AddWatchMode.PERSISTENT_RECURSIVE:
+                            client._persistent_recursive_watchers[
+                                request.path
+                            ].add(watcher)
```

Comment:
I wonder if this might benefit from an `else` clause?

I realize it's a no-op, but it still took me a moment to realize that's because it might be a non-persistant watch that's being added... so might be easier for a reader to understand with a code comment... 

But code comments can go out of sync, so a simple assertion that it's the expected mode might be useful like:
```python
else:
    if request.mode not in [expected modes]:
       raise <relevant exception>
```

that way both clarifies the code and also catches stupid errors if somehow a case was missed down the road.

Alternatively, this might read even simpler as a switch statement with a default case... but I realize those require Python 3.10 so maybe leave a:
```python
# TODO: switch these if/else clauses to switch statements where appropriate once we drop Python 3.9 support
```